### PR TITLE
[move] Remove `Compiled{Module,Script}Mut`

### PR DESCRIPTION
**Pull request by modocache**
`CompiledScript` and `CompiledModule` were meant to be bounds-checked analogues of `CompiledScriptMut` and `CompiledModuleMut`, but I've heard from some people on the team that they may have outlived their usefulness (and indeed the bounds checker has already been modified to operate on a non-`Mut` `Compiled{Script,Module}`).

This pull request removes the distinction.

### CODE REVIEW

File: language/move-binary-format/src/file_format.rs
Comment (tnowacki) [inline language/move-binary-format/src/file_format.rs:1650]: I know it was sort of the whole point of this migration, but when I saw this PR I was a bit skittish of seeing everything being mutable. It's one thing to say we want it done and another to actually see it lol

But after rethinking about it, I think most uses of a CompiledModule are through `&CompiledModule` so the mutation isn't an issue, and it will be nice to maybe slowly kill off the `ModuleAccess` trait.

Any additional thoughts @vgao1996? Feel free to approve it once you take a look

**Review by tnowacki — COMMENTED** [review]
Bit odd seeing freeze around still. I'm guessing that will go away in a subsequent PR?

**Comment by modocache** [comment]
Yeah! I was thinking of replacing it with direct calls to the bounds checker, or renaming it to `verify_bounds` or something.

**Comment by tnowacki** [comment]
> Yeah! I was thinking of replacing it with direct calls to the bounds checker, or renaming it to `verify_bounds` or something.
> 
> EDIT: I just rebased on top of `main` and was considering also removing or renaming `freeze`, but I think it'll be a large diff, so I think I will just stick with this changeset for now.

You might want to also stick it into `verify_module` and `verify_script` in `language/bytecode-verifier/src/verifier.rs`

File: language/move-binary-format/src/file_format.rs
Comment (modocache) [reply language/move-binary-format/src/file_format.rs:1687]: Ah, looks like @vgao1996 is on PTO until next week. Up to you @tnowacki whether to wait until Monday for this to land or to accept and have his review ex post facto.

Comment (tnowacki) [reply language/move-binary-format/src/file_format.rs:1687]: Let's just land it. I think this will all feel better as we clean up more and more. I'd love to see the `ModuleAccess` trait die and everything in `language/move-binary-format/src/views.rs` die or get moved into the prover 

This was a cleanup long planned by Dario and myself, so should be fun to pick away at it (I might even pitch in just for the fun so we should sync up)

**Review by tnowacki — DISMISSED** [review]

**Pull request closed without merging**
